### ISSUE TITLE
Tabs in ObsEdit always re-mount

### ISSUE THREAD

**Issue by jtklein** [inaturalist/iNaturalistReactNative#403]
**Describe the bug**
On iOS (because on Android the map does not work), in the ObsEdit screen every time the Data tab is opened, even when it was already opened before in the same page view, the whole tab get's initialized again, so, e.g., the map in the tab starts with the loading indicator. I think we should try to have the tab initialise on the first render only, and then keep it available in the page for re-access. This was brought up in PR https://github.com/inaturalist/iNaturalistReactNative/pull/385

**To Reproduce**
1. Select an observation
2. Tap on "Data" tab
3. See that the map initializes with loading indicator
4. Tap the "Activity" tab and back the "Data" tab
5. See that the map initializes again

**Expected behavior**
The map should stay on the tab and be immediately available when I tap back there

**Context (please complete the following information):**
Device: iPhone 7

**Additional context**
In ObsEdit render there is a ternary that shows either the data tab or activity tab, keeping a reference for those tabs outside might help to fix this.

**Comment by Chrischuck** [comment]
One solution is using css to hide the component and technically maintain the component on the DOM.

I propose a reusable component something like so:

```
const HideView = ( { noInitialRender = false, hidden = true, children } ): React.Node => {
    const [rendered, setRendered] = useState( false );

    useEffect( () => {
      if ( !hidden ) {
        setRendered( true );
      }
    }, [hidden] );

    if ( noInitialRender && !rendered && hidden ) {
      return null;
    }

    const props = hidden ? { className: "hidden " } : {};
    return <View {...props}>{children}</View>;
  };

export default HideView;
```

and on the consumer end can be used simply:

```
<HideView hidden={currentTabId === someTab}>
    <Text>Hidden!</Text>
</HideView>
```

The `noInitialRender` lets us either render the component "in the background" (it will block, so be careful), or lets use render only when the tab is unhidden.

If this is an agreed upon solution, I'd be happy to implement this.

Should note that using this component to hide components has the benefit in preventing re-initialization of components, I feel it should be used sparingly. IMO it's better to just take components off the DOM.

**Comment by albullington** [comment]
Closed in #412
